### PR TITLE
[JAX] Support single-process-multi-device EP via XLA-borrowed communicator

### PR DESCRIPTION
**Pull request by phu0ngng**
# Description
Support single-process, multi-device JAX EP using XLA-borrowed NCCL communicators.

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refactoring

## Changes
- Manage EP initialization, state, and teardown per device.
- Size-handle buffers across local devices and reject undersized buffers.
- Update bootstrap, launchers, examples, and tests for multi-device execution.

## Testing
- EP guard/grouping tests: 7 passed, 1 skipped.
- All pre-commit checks passed.

# Checklist:

- [x] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [x] The functionality is complete
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: tests/jax/test_multi_process_ep.py [hunk 1]
```diff
@@ -1065,19 +1201,84 @@ def test_force_on_without_build_raises(self):
             self._use("1", built=False, supported=True)
 
 
+class TestEpBootstrapMultiDeviceGuard(unittest.TestCase):
+    """Require borrowed XLA communicators for multiple local devices."""
+
+    def test_raises_when_borrowed_comm_unavailable(self):
+        import transformer_engine.jax.ep as ep_mod
+
+        with mock.patch.object(ep_mod.jax, "local_device_count", return_value=2), mock.patch.object(
```
Comment (jberchtold-nvidia) [inline tests/jax/test_multi_process_ep.py:1210]: I like this mocking of tests to test our error paths too!

**Review by jberchtold-nvidia — DISMISSED** [review]
Looks great! Thanks!

Once CI passes, all good to merge from my side

**Review by jberchtold-nvidia — DISMISSED** [review]

**Review by jberchtold-nvidia — APPROVED** [review]

**Comment by phu0ngng** [comment]
CI pass except L1_jax_distributed_unittest--B200_8GPU softmax failed, which is unrelated to this PR.
